openedx/adg/lms/applications/tests/test_api_views.py

```diff
@@ -314,3 +325,38 @@ def test_work_experience_api_delete(work_experience_data, login_client):
     url = reverse('applications_api:work_experience-detail', kwargs={'pk': work_experience.id})
     response = login_client.delete(url)
     assert response.status_code == status.HTTP_204_NO_CONTENT
+
+
+@pytest.mark.django_db
+def test_update_is_work_experience_not_applicable_api_with_true(
+    work_experience_data,
+    work_experience_not_applicable_api_data,
+    login_client
+):
+    """
+    Test the is_work_experience_not_applicable_api with true value
+    """
+    is_work_experience_not_applicable_api_data = work_experience_not_applicable_api_data('true')
+    url = reverse('applications:work_experience-update_is_not_applicable')
+    response = login_client.patch(
+        url, data=json.dumps(is_work_experience_not_applicable_api_data), content_type=JSON_CONTENT_TYPE
+    )
+    assert response.status_code == status.HTTP_200_OK
+    assert WorkExperience.objects.count() == 0
+
+
+@pytest.mark.django_db
+def test_update_is_work_experience_not_applicable_api_with_false(
+    user_application,
+    work_experience_not_applicable_api_data,
+    login_client
+):
+    """
+    Test the is_work_experience_not_applicable_api with false value
+    """
+    is_work_experience_not_applicable_api_data = work_experience_not_applicable_api_data('false')
+    url = reverse('applications:work_experience-update_is_not_applicable')
+    response = login_client.patch(
+        url, data=json.dumps(is_work_experience_not_applicable_api_data), content_type=JSON_CONTENT_TYPE
+    )
+    assert response.status_code == status.HTTP_200_OK
```

Comment:
These two test cases are exactly same except for the parameter being passed to 'work_experience_not_applicable_api_data' hence, using parametrisation these two test cases can be combined

Reply:
Both tests aren't exactly same. 

for api with true we need to create work experience in database and after calling api we need to verify these work experiences are deleted successfully. but in case of api with false we only need to check its success.

Reply:
But I have removed the redundancy using fixtures.